pkg/prometheus/operator.go

```diff
@@ -455,12 +480,48 @@ func (c *Operator) Run(ctx context.Context) error {
 	if err := c.waitForCacheSync(ctx); err != nil {
 		return err
 	}
+
+	// Refresh the status of the existing Prometheus objects.
+	_ = c.promInfs.ListAll(labels.Everything(), func(obj interface{}) {
+		c.addToStatusQueue(obj)
+	})
+
 	c.addHandlers()
 
 	if c.kubeletSyncEnabled {
 		go c.reconcileNodeEndpoints(ctx)
 	}
 
+	// Run a goroutine that refreshes regularly the Prometheus objects that
+	// aren't available.
+	// This is a brute-force approach to ensure that the Prometheu status
+	// conditions reflect the current state of the world.
+	go func() {
```

Comment:
I see, but Shouldn't the operator react to status updates for the StatefulSet as well?

Reply:
the operator reacts on StatefulSet updates but the StatefulSet object itself is updated when the init-config-reloader container runs but not after. IIUC the StatefulSet controller only updates the status subresource when the number of ready/updated replicas changes which doesn't happen until all containers from the pod are ready.

Reply:
> Do you think it might make sense to watch the pods directly, or would that be too much overhead?

It would be a legit improvement from my POV but 
1) the PR is already quite large.
2) the operator would need new permissions on pod resources so it should be an opt-in probably (e.g. watch pods and degrade to "poller" mode if it fails).

Reply:
I agree, it would be a more involved change. We can keep it as it is now and make further improvements down the road if needed.